internal/elasticsearch/index/index.go

```diff
@@ -37,6 +105,301 @@ func ResourceIndex() *schema.Resource {
 				validation.StringMatch(regexp.MustCompile(`^[a-z0-9!$%&'()+.;=@[\]^{}~_-]+$`), "must contain lower case alphanumeric characters and selected punctuation, see: https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html#indices-create-api-path-params"),
 			),
 		},
+		// Static settings that can only be set on creation
+		"number_of_shards": {
+			Type:        schema.TypeInt,
+			Description: "Number of shards for the index. This can be set only on creation.",
+			ForceNew:    true,
+			Optional:    true,
+		},
+		"number_of_routing_shards": {
+			Type:        schema.TypeInt,
+			Description: "Value used with number_of_shards to route documents to a primary shard. This can be set only on creation.",
+			ForceNew:    true,
+			Optional:    true,
+		},
+		"codec": {
+			Type:        schema.TypeString,
+			Description: "The `default` value compresses stored data with LZ4 compression, but this can be set to `best_compression` which uses DEFLATE for a higher compression ratio. This can be set only on creation.",
+			ForceNew:    true,
+			Optional:    true,
+		},
+		"routing_partition_size": {
+			Type:        schema.TypeInt,
+			Description: "The number of shards a custom routing value can go to. This can be set only on creation.",
+			ForceNew:    true,
+			Optional:    true,
+		},
+		"load_fixed_bitset_filters_eagerly": {
+			Type:        schema.TypeBool,
+			Description: "Indicates whether cached filters are pre-loaded for nested queries. This can be set only on creation.",
+			ForceNew:    true,
+			Optional:    true,
+		},
+		"shard_check_on_startup": {
+			Type:        schema.TypeString,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Description: "Whether or not shards should be checked for corruption before opening. When corruption is detected, it will prevent the shard from being opened. Accepts `false`, `true`, `checksum`.",
+			ForceNew:    true,
+			Optional:    true,
+		},
+		"sort_field": {
+			Type:        schema.TypeSet,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Description: "The field to sort shards in this index by.",
+			ForceNew:    true,
+			Optional:    true,
+		},
+		"sort_order": {
+			Type:        schema.TypeList,
```

Comment:
Yes this is intentional because when `sort_field` has multiple fields, `sort_order` can have dup strings like `["asc", "asc"]`.
Let me add a comment!